eelsdbconverter/metainfo.py

```diff
@@ -0,0 +1,113 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+from nomad.metainfo import Section, Quantity, MSection, SubSection
+import numpy as np
+
+from nomad.datamodel import EntryArchive
+from nomad.metainfo.metainfo import Datetime
+
+class DeviceSettings(MSection):
+    device_name = Quantity(type=str)
+    max_energy = Quantity(type=str)
+    min_energy = Quantity(type=str)
+    guntype = Quantity(type=str)
+    beam_energy = Quantity(type=str)
+    resolution = Quantity(type=str)
+    step_size = Quantity(type=str)
+    acquisition_mode = Quantity(type=str)
+    beam_current = Quantity(type=str)
+    detector_type = Quantity(type=str)
+    dark_current=Quantity(type=str)
+
+
+class Sample(MSection):
+    sample_id = Quantity(type=str)
+    sample_title = Quantity(type=str)
+    formula = Quantity(type=str)
+    elements = Quantity(type=str, shape=['*'])
+
+class Experiment(MSection):
+    method_name = Quantity(type=str)
+    method_abbreviation = Quantity(type=str)
+    experiment_id = Quantity(type=str)
+    experiment_publish_time = Quantity(
+        type=Datetime, description='The datetime when this experiment was published.')
+    experiment_start_time = Quantity(
+        type=Datetime, description='The datetime of the beginning of the experiment.')
+    experiment_end_time = Quantity(
+        type=Datetime, description='The datetime of the experiment end.')
+    edges = Quantity(type=str, shape = ['*'])
+
+
+class Instrument(MSection):
+    source_label = Quantity(type=str)
+    section_device_settings = SubSection(sub_section=DeviceSettings, repeats=True)
+
+class AuthorGenerated(MSection):
+    permalink = Quantity(type=str)
+    author_name = Quantity(type=str)
+    author_profile_url = Quantity(type=str)
+    author_profile_api_url = Quantity(type=str)
+
+    description = Quantity(type=str)
+
+    group_name = Quantity(type=str)
+
+class DataHeader(MSection):
+    # channel_id = Quantity(type=str)
+    label = Quantity(type=str)
+    unit = Quantity(type=str)
+
+# class NumericalValues(MSection):
+    # data_values = Quantity(type=np.dtype(np.float64), shape=['*'])
+
+class Spectrum(MSection):
+  n_values = Quantity(type=int)
+  energy = Quantity(type=np.dtype(np.float64), shape=['n_values'], unit='eV', description='This column contains the range of energies of the spectrum')
+  count = Quantity(type=np.dtype(np.float64), shape=['n_values'], unit=' ', description='The count at each energy value, dimensionless')
+
+class Metadata(MSection):
+    section_sample = SubSection(sub_section=Sample, repeats=True)
+    section_experiment = SubSection(sub_section=Experiment, repeats=True)
+    section_instrument = SubSection(sub_section=Instrument, repeats=True)
+    section_author_generated = SubSection(sub_section=AuthorGenerated, repeats=True)
+    section_data_header = SubSection(sub_section=DataHeader, repeats=True)
+
+class Data(MSection):
+    repository_name = Quantity(
```

Comment:
Maybe we need a metadata sections for data? Everything here except section_spectrum is not really "the data", but metadata about how to find it, preview it, etc.

Reply:
That was the whole point to split the whole measurement into data and metadata. section_spectrum is the numerical data, but everything else is the metadata to this data.

I'm not sure if I follow your point here.

Reply:
The quantities `repository_url`, `preview_url`, `entry_repository_url`, `repository_name` are not really data and should not be in the section data. You define section data to hold the actual data. The original intend behind this section was different and more like "metadata that is data related". Therefore these could/should move somewhere else. Maybe `measurement.metadata.origin` or something. `preview_url` might be arguable if this actually belongs in data?

Reply:
Yeah, I agree. I missed the point earlier. I could move these currently to `AuthorGenerated` because very loosely the database in this case is the source of the data. And these are miscellaneous field? I'm a bit hesitant to add more new subsections. 
Or would you still recommend a different section header? Like `measurement.metadata.origin`

Reply:
Again, i don't like `AuthorGenerated` as a section. It is fine to have `Author` (when the data describes the Authors, name, affiliation, etc.). Otherwise, stuff in AuthorGenerated should move to where it thematically belongs. That a quantity is generated by a human, can be marked otherwise (e.g. via Metainfo Category). 

Following this, I would suggest a different section, like `origin`. The only questionable quantity is `preview_url`. I guess you find arguments for having this in `measurement.data`, or `measurment.metadata`, or `metasurement.metadata.origin`?

Reply:
The reason to have `AuthorGenerated` as a prominent section is because this is the section where quantities such as `internal_project_id`, `internal_sample_id `would be expected to go. The other items in the list are expected to be human generated, such as `author_notes`. 
Currently, we need a section like `origin` as you have pointed out because we're reading off data from a database, but is that expected to be a usual case? If that is expected to be the case, then I can create a section like `origin`.

Reply:
We are discussing now two independent topics. 

There will be different types of origins: external databases (e.g. EELS), an oasis (e.g. XPSHub), an 3rd party repository (e.g. Xenodo). There are probably more. Consider that you might only have the metadata at hand. Maybe we even need subsections to `measurment.metadata.origin`. 

I dislike the idea of having a plain bucket for everything that an author generates. The sections should describe a common theme for what their quantities are about and not how they were obtained. If it is a sample id it belongs in sample, if it is an author note, its either a note about the sample, instrument, or the measurement in general. What if the sample id is not "author generated"? Building the containment hierarchy on different criteria (topic and method of acquisition) raises questions about where to put quantities. E.g, does sample id belong in sample or author generated?

We could generalize and add certain common quantities to all metadata sections (internal_id, comment, references)?

Reply:
I agree to the points you made. I'll create a new section `origin` and adding certain quantities to all Metadata sounds like the best solution. I'll change the metainfo.py to reflect these.